src/App.js

```diff
@@ -43,8 +50,9 @@ class ColorApp extends Component {
     })
     
     this.setState({
-      color: colors
- }
+      colors: colors
+   })
+  }
   
```

Comment:
Line 52 was the root of your problem. Your state was not changing when the toggle button was clicked and it was due to the name of the state. I changed the name of your state form colors to color so Im actually not sure if i cause this to break or not. NOTE: moving forward if always check to see if your state is changing. This is easy. Just place a console log where you expect that action to happen at. in this  example I console logged the state the index and the iteration(i) to make sure I was seeing the correct results.